### PR TITLE
FilterValuePicker tests

### PR DESCRIPTION
**Pull request by ranquild**

### CODE REVIEW

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+          field_id: ORDERS.QUANTITY,
+          values: [
+            [10, "To-do"],
+            [20, "In-progress"],
+            [30, "Completed"],
+          ],
+        }),
+      });
+      expect(screen.getByRole("checkbox", { name: "To-do" })).toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "To-do" })).toBeChecked();
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx:469]: Did you mean the second line to assert that "In-progress" is not checked?

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+    it("should not allow to add invalid values", async () => {
+      const { onChange, onFocus, onBlur } = await setupNumberPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+      });
+
+      const input = screen.getByPlaceholderText("Enter a number");
+      userEvent.type(input, "abc");
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx:527]: I think it's worth asserting that whitespace-only string will be rejected as well.
I remember the initial implementation would treat `" "` as `0`.

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+
+      userEvent.click(screen.getByText("In-progress"));
+      expect(onChange).toHaveBeenCalledWith([10, 20]);
+    });
+  });
+
+  describe("no values", () => {
+    const column = findColumn("PEOPLE", "PASSWORD");
+
+    it("should allow to add a value", async () => {
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx:483]: I'd also assert handling non-integer values

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,535 @@
...
+      const { onChange } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: ["abc", "bce"],
+      });
+
+      userEvent.type(screen.getByLabelText("Filter value"), "{backspace}");
+
+      expect(onChange).toHaveBeenLastCalledWith(["abc"]);
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx:414]: I think it's worth asserting if it's possible to clear the value:

```suggestion
      expect(onChange).toHaveBeenLastCalledWith(["abc"]);
      
      userEvent.type(screen.getByLabelText("Filter value"), "{backspace}");

      expect(onChange).toHaveBeenLastCalledWith([]);
```

**Review by kamilmielnik — APPROVED** [review]

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx:561]: Fixed

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx:517]: Fixed

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx]: Fixed

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx (hunk 4 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx:433]: Fixed

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,535 @@
...
+          field_id: PRODUCTS.CATEGORY,
+          values: [
+            ["t", "To-do"],
+            ["p", "In-progress"],
+            ["c", "Completed"],
+          ],
+        }),
+      });
+      expect(screen.getByRole("checkbox", { name: "To-do" })).toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "To-do" })).toBeChecked();
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx]: Fixed

**Pull request merged**